http.cc

```diff
@@ -81,12 +82,27 @@ int trace_send_response(struct pt_regs *ctx) {
   events.perf_submit(ctx, &ev, sizeof(ev));
   return 0;
 }
+
+int trace_send_response_header(struct pt_regs *ctx) {
+  struct event_t ev = { .type = HTTP_EVENT_SEND_RESP_HDR };
+  void *pos = NULL;
+  bpf_usdt_readarg(1, ctx, &ev.conn_id);
+  bpf_usdt_readarg(2, ctx, &ev.req_id);
+  bpf_usdt_readarg(3, ctx, &pos);
+  bpf_usdt_readarg(4, ctx, &ev.header.name_len);
+  bpf_probe_read(&ev.header.name, sizeof(ev.header.name), pos);
+  bpf_usdt_readarg(5, ctx, &pos);
+  bpf_usdt_readarg(6, ctx, &ev.header.value_len);
+  bpf_probe_read(&ev.header.value, sizeof(ev.header.value), pos);
+  events.perf_submit(ctx, &ev, sizeof(ev));
+  return 0;
+}
 )";
 
 #define MAX_HDR_LEN 128
 #define MIN(x, y) (((x) < (y)) ? (x) : (y))
 
-enum { HTTP_EVENT_RECEIVE_REQ, HTTP_EVENT_RECEIVE_REQ_HDR, HTTP_EVENT_SEND_RESP };
+enum { HTTP_EVENT_RECEIVE_REQ, HTTP_EVENT_RECEIVE_REQ_HDR, HTTP_EVENT_SEND_RESP, HTTP_EVENT_SEND_RESP_HDR };
```

Comment:
FYI: a BPF module can include C headers so you don't need to copy type definitions.

Reply:
Nice, thanks for the great tip. I think using `data-types.h` would fit nicely for that approach. It's going to require some tweaks, so I'll submit a separate PR for that.